src/components/NetworkBN/index.js

```diff
@@ -16,98 +16,102 @@ import {
 } from 'selectors';
 import { nodePropTypes, networkPropTypes, inferenceResultsPropTypes } from 'models';
 import { getArrowsPositions } from 'utils/arrows-positions';
-import Network, { ContextMenuType } from '../Network';
+import Network from '../Network';
 
 import AddNodeModal from '../AddNodeModal';
 import EditCptModal from '../EditCptModal';
 import EditStatesModal from '../EditStatesModal';
 
 class NetworkBN extends Component {
-  constructor(props) {
-    super(props);
-    const { dispatch } = props;
+  state = {
+    key: 1,
+    editingNodeStates: null,
+    editingNodeCpt: null,
+  };
+
+  componentDidMount() {
+    window.addEventListener('keyup', this.handleKeyup);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keyup', this.handleKeyup);
+  }
+
+  get nodes() {
+    const { inferenceResults, network, nodes } = this.props;
+
+    return nodes.map(node => ({
+      ...node,
+      name: node.id,
+      results: inferenceResults[node.id],
+      belief: network.beliefs[node.id],
+    }));
+  }
 
-    this.state = {
-      key: 1,
-      editingNodeStates: null,
-      editingNodeCpt: null,
-    };
+  get arrowContextItems() {
+    const { dispatch } = this.props;
```

Comment:
Do not use dispatch directly as a component prop. I hard recommend to use the `mapDispatchToProps` react-redux feature instead. This kind of approach will make your component dependent to a `dispatch` prop when actually what you need is the function/callback which will call the needed action. Your component doesn't need a dispatch function it needs an action callback.